modules/data-entry/src/main/frontend/src/questionnaire/SubjectSelector.jsx

```diff
@@ -235,9 +250,206 @@ function UnstyledSelectParentDialog (props) {
 
 export const SelectParentDialog = withStyles(QuestionnaireStyle, {withTheme: true})(UnstyledSelectParentDialog)
 
+// The value of a subjectType's parents are either an array, or if it is length 1 it will just be an object
+// We must cast each case into an array to handle it properly
+export const parseToArray = (object) => {
+  // Null or undefined is length 0
+  if (!object) {
+    return [];
+  }
+
+  // A non-array is length 1
+  if (!Array.isArray(object)) {
+    return [object];
+  } else {
+    return object;
+  }
```

Comment:
This if statement can probably be made more concise:
```suggestion
  } return (!Array.isArray(object)) ? [object] : object;
```

Reply:
Good catch, patched in `7ac078a` :heavy_check_mark: